rct_ros_tools/include/rct_ros_tools/target_finder_plugin.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <ros/node_handle.h>
+#include <string>
+#include <rct_image_tools/target.h>
+#include <rct_image_tools/target_finder.h>
+
+namespace rct_ros_tools
+{
+/**
+ * @brief Target finder plugin that can be initialized by file or XmlRpcValue (i.e. from a ROS parameter)
+ */
+class TargetFinderPlugin : public rct_image_tools::TargetFinder
+{
+public:
+  using TargetFinder::TargetFinder;
+
+  rct_image_tools::TargetFeatures findTargetFeatures(const cv::Mat& image) const override
+  {
+    return finder_->findTargetFeatures(image);
+  }
+
+  cv::Mat drawTargetFeatures(const cv::Mat& image,
+                             const rct_image_tools::TargetFeatures& target_features) const override
+  {
+    return finder_->drawTargetFeatures(image, target_features);
+  }
+
+  const rct_image_tools::Target& target() const override
+  {
+    return finder_->target();
+  }
+
+  virtual void init(const XmlRpc::XmlRpcValue& config) = 0;
+  virtual void init(const std::string& file) = 0;
```

Comment:
Might be better to drop the XmlRpc option and only use yaml which would make this easier to move to ROS2. A simple XmlRpc -> yaml converter could be provided for the parameter server use case.

Reply:
So maybe we unify on `YAML::Node` instead?

Reply:
That seems reasonable to me.